source/module_elecstate/elecstate_print.cpp

```diff
@@ -371,6 +385,17 @@ void ElecState::print_etot(const bool converged,
                 std::cout << std::setw(10) << get_ucell_tot_magnetization();
                 std::cout << std::setw(10) << get_ucell_abs_magnetization();
             }
+
+#ifdef __RAPIDJSON
+            //add Json of scf mag
+            Json::add_output_scf_mag(
+                get_ucell_tot_magnetization(), get_ucell_abs_magnetization(),
+                this->f_en.etot * ModuleBase::Ry_to_eV,
+                (this->f_en.etot - this->f_en.etot_old) * ModuleBase::Ry_to_eV,
+                scf_thr,
+                duration
+            );
+#endif //__RAPIDJSON 
```

Comment:
Should add these `add_output_scf_mag` function after 
https://github.com/deepmodeling/abacus-develop/blob/2f5217f847be79eedac71413b213e43ea46981da/source/module_esolver/esolver_ks.cpp#L456-L467

Reply:
fix that! Now json format of output part is as below:
```
    "output": [
        {
            "e_fermi": 6.550740286118238,
            "energy": -215.50569837287007,
            "scf_converge": true,
            "force": [
                [
                    0.0,
                    0.0,
                    0.0
                ],
                [
                    0.0,
                    0.0,
                    0.0
                ]
            ],
            "stress": [
                [
                    7.9161361066141249,
                    0.0,
                    0.0
                ],
                [
                    0.0,
                    7.9161361066141249,
                    0.0
                ],
                [
                    0.0,
                    0.0,
                    7.9161361066141249
                ]
            ],
            "coordinate": [
                [
                    0.0,
                    0.0,
                    0.0
                ],
                [
                    2.55,
                    2.55,
                    2.55
                ]
            ],
            "mag": [
                0.0,
                0.0
            ],
            "cell": [
                [
                    0.5,
                    0.5,
                    0.0
                ],
                [
                    0.5,
                    0.0,
                    0.5
                ],
                [
                    0.0,
                    0.5,
                    0.5
                ]
            ],
            "total_mag": 0.0,
            "absolute_mag": 0.0,
            "scf": [
                {
                    "energy": -215.46046192299714,
                    "ediff": 0.0,
                    "drho": 0.06839358317524727,
                    "time": 1.876413259
                },
                {
                    "energy": -215.50350891967273,
                    "ediff": -0.04304699667560524,
                    "drho": 0.0018936564358656078,
                    "time": 0.600312733
                },
                {
                    "energy": -215.5055135491307,
                    "ediff": -0.0020046294579900848,
                    "drho": 0.00002315722527038331,
                    "time": 0.7927257230000002
                },
                {
                    "energy": -215.5056964119524,
                    "ediff": -0.00018286282169231044,
                    "drho": 7.934694501887137e-7,
                    "time": 0.8368760239999999
                },
                {
                    "energy": -215.50569837287007,
                    "ediff": -0.0000019609176464104156,
                    "drho": 1.527851346969937e-8,
                    "time": 0.8225977319999993
                }
            ]
        }
    ]

```

Reply:
It seems that the function `add_output_scf_mag` has not been moved to abacus-develop/source/module_esolver/esolver_ks.cpp

Reply:
Oh! It seems that the last push didn't work, and now the push is correct. By the way, `add_output_scf_mag` after the last if causes an `scf` generation error. When you put it in `printiter`, it generates the `scf` array correctly (so far it's written this way).